core/src/tauri/command.rs

```diff
@@ -358,6 +360,77 @@ pub async fn rebase(state: tauri::State<'_, State>) -> Result<(), TauriError> {
     Ok(())
 }
 
+#[tauri::command]
+pub async fn acquire_locks(
+    state: tauri::State<'_, State>,
+    files: Vec<ModifiedFile>,
+    fail_if_locked: bool,
+) -> Result<bool, TauriError> {
+    // get github username
+    let mut res = state
+        .client
+        .get(format!("{}/repo/gh/user", state.server_url))
+        .send()
+        .await?;
+
+    if res.status().is_client_error() {
+        let body = res.text().await?;
+        return Err(TauriError { message: body });
+    }
+
+    let username = match res.json::<UserInfoResponse>().await {
+        Ok(res) => res.username,
+        Err(err) => {
+            return Err(TauriError {
+                message: format!("Failed to get github username: {}", err),
+            });
+        }
+    };
+
+    // only lock files that are unlocked
+    let mut paths: Vec<String> = Vec::new();
```

Comment:
We should move this defensive logic into the LockRequest, ideally even if a lock request comes from another source, such as Unreal, it would perform all the same checks.